src/main/groovy/org/shipkit/internal/gradle/plugin/PluginUtil.java

```diff
@@ -0,0 +1,33 @@
+package org.shipkit.internal.gradle.plugin;
+
+import org.gradle.api.Project;
+import org.gradle.api.file.FileTree;
+import org.gradle.api.plugins.JavaPluginConvention;
+import org.gradle.api.tasks.SourceSet;
+import org.gradle.api.tasks.util.PatternSet;
+
+import java.io.File;
+import java.util.Set;
+
+class PluginUtil {
+    static final String DOT_PROPERTIES = ".properties";
+
+    static Set<File> discoverGradlePluginPropertyFiles(Project project) {
+        FileTree resources = getMainSourceSet(project).getResources();
+        return getFilteredFileset(resources, "META-INF/gradle-plugins/*" + DOT_PROPERTIES);
+    }
+
+    static Set<File> discoverGradlePlugins(Project project) {
+        FileTree allJava = getMainSourceSet(project).getAllJava();
+        return getFilteredFileset(allJava, "**/*Plugin.java", "**/*Plugin.groovy");
+    }
+
+    private static SourceSet getMainSourceSet(Project project) {
```

Comment:
This is a nice utility method! If new places in the code we already get the main source set this way. Can you refactor those instances so that we are using the utility method everywhere? Also, this method does not seem the be related to plugins, perhaps we can move it to a more generic utility class?